src/NServiceBus.Core/InitializableEndpoint.cs

```diff
@@ -105,6 +119,46 @@ TransportInfrastructure InitializeTransport()
             return transportInfrastructure;
         }
 
+        ReceiveConfiguration BuildReceiveConfiguration(TransportInfrastructure transportInfrastructure)
+        {
+            var receiveConfiguration = ReceiveConfigurationBuilder.Build(settings, transportInfrastructure);
+
+            if (receiveConfiguration == null)
+            {
+                return null;
+            }
+
+            //note: remove once settings.LogicalAddress() , .LocalAddress() and .InstanceSpecificQueue() has been obsoleted
+            settings.Set<ReceiveConfiguration>(receiveConfiguration);
+
+            return receiveConfiguration;
+        }
+
+        ReceiveComponent CreateReceiveComponent(ReceiveConfiguration receiveConfiguration, TransportInfrastructure transportInfrastructure, IPipelineCache pipelineCache, EventAggregator eventAggregator)
+        {
+            var mainPipeline = new Pipeline<ITransportReceiveContext>(builder, pipelineConfiguration.Modifications);
+            var mainPipelineExecutor = new MainPipelineExecutor(builder, eventAggregator, pipelineCache, mainPipeline);
+            var errorQueue = settings.ErrorQueueAddress();
+            var queueBindings = settings.Get<QueueBindings>();
+            var receiveInfrastructure = transportInfrastructure.ConfigureReceiveInfrastructure();
```

Comment:
This call means that `ConfigureReceiveInfrastructure` is now always being called, but before it was not called if the endpoint was send-only.

Reply:
I've added acceptance tests for the full startup sequence, fixed the bug and verified that its now identical compared to the current develop branch

https://github.com/Particular/NServiceBus/pull/5011/commits/87853b87d6c4fc2a1aaf3d0a2fb36ba35f98a210